### ISSUE TITLE
Cannot use @UntilDestroy on components in unit tests

### ISSUE THREAD

**Issue by euangoddard** [ngneat/until-destroy#51]
When unit-testing components decorated with `@UntilDestroy` this decorator has no effect since at present the JIT compiler is used for unit tests in Angular and not the AOT compiler. As a result the private properties that this decorator relies on don't exist on the components and our components behave differently under testing scenarios to real-life scenarios.

I note that in your unit tests, you do not use the real `@Component` annotation, but create the scenario that the AOT compiler uses. Having experimented with the code, I think it's possible to simply set properties directly on the target in the decorator in JIT mode to overcome this.

**Comment by PabloDotcom** [comment]
I've done migration from **ngx-take-until-destroy** to **ngneat/until-destroy** (after migration to angular packages inver. 9.0.0). As application itself works fine, I've got issues with jest unit tests.

Error:
TypeError: Cannot read property 'onDestroy' of undefined

Somehow related?

**Comment by NetanelBasal** [comment]
@euangoddard our tests passes https://github.com/ngneat/until-destroy/blob/master/integration/app/app.component.spec.ts

**Comment by euangoddard** [comment]
@NetanelBasal this is because your components are not real Angular components created with the `@Component` decorator. You are synthesising the extra properties that the Angular compiler adds on during the build.

**Comment by arturovt** [comment]
@euangoddard would it be possible to create a reproduction for us? I have a limited amount of time and I would not want to try to reproduce the problem myself.

**Comment by euangoddard** [comment]
I will generate a new CLI project and see if there's something obvious

**Comment by euangoddard** [comment]
I've made a sample project with a timer and have been checking whether it continues after the component is destroyed, but it seems to stop. I'm also not getting the error that @PabloDotcom reported which I have seen before. I can't recall the tests that were breaking in my codebase as I worked around the issue in all cases. If I can locate them I'll revert my workaround as maybe the final version of Angular has fixed this. Alternatively I might be missing a setting in angular.json since the current project has been upgraded all the way from Angular 4!

I think it would be good for your test suite to use real components rather than making something that looks like what the compiler does to check that the decorator really works in all situations
